webapi/src/main/scala/org/knora/webapi/responders/v1/ResourcesResponderV1.scala

```diff
@@ -374,21 +374,23 @@ final case class ResourcesResponderV1Live(
                                    )
 
       // Make sure the user has permission to see the start node.
-      _ = if (
-            PermissionUtilADM
-              .getUserPermissionV1(
-                entityIri = startNode.nodeIri,
-                entityCreator = startNode.nodeCreator,
-                entityProject = startNode.nodeProject,
-                entityPermissionLiteral = startNode.nodePermissions,
-                userProfile = userProfileV1
-              )
-              .isEmpty
-          ) {
-            throw ForbiddenException(
+      permission =
+        PermissionUtilADM
+          .getUserPermissionV1(
+            entityIri = startNode.nodeIri,
+            entityCreator = startNode.nodeCreator,
+            entityProject = startNode.nodeProject,
+            entityPermissionLiteral = startNode.nodePermissions,
+            userProfile = userProfileV1
+          )
+      _ <-
+        ZIO
+          .fail(
+            ForbiddenException(
               s"User ${graphDataGetRequest.userADM.id} does not have permission to view resource ${graphDataGetRequest.resourceIri}"
             )
-          }
+          )
+          .when(permission.isEmpty)
```

Comment:
is it correct that `permission` isn't used afterwards? then this could be reduced to one `_ <- ZIO.fromOption(...)`